src/core/thread/mesh_forwarder.cpp

```diff
@@ -890,11 +916,19 @@ ThreadError MeshForwarder::SendMacDataRequest(void)
     message = mNetif.GetIp6().mMessagePool.New(Message::kTypeMacDataPoll, 0);
     VerifyOrExit(message != NULL, error = kThreadError_NoBufs);
 
-    SuccessOrExit(error = SendMessage(*message));
-    otLogInfoMac("Sent poll");
+    error = SendMessage(*message);
 
-    // restart the polling timer
-    mPollTimer.Start(mPollPeriod);
+    if (error == kThreadError_None)
+    {
+        otLogInfoMac("Sent poll");
+
+        // restart the polling timer
+        mPollTimer.Start(mPollPeriod);
+    }
+    else
+    {
+        mNetif.GetIp6().mMessagePool.Free(message);
```

Comment:
This is fine but I think we can call `Free()` on the `Message` object directly, as in `message->Free()`. Any reason for not using this?